File: Client/Battleship/src/Models/Cell.js
class Cell {

    static State = {
        Attacked: 'Attacked',
        Hit: 'Hit',
        Neutral: 'Neutral'
    }

    constructor(index) {
        this.index = index;
        this.state = Cell.State.Neutral;
    }

    registerAttack(didHit) {
        this.state = didHit ? Cell.State.Hit : Cell.State.Attacked;
    }
}

File: BattleshipServer/Sources/App/Game/Cell.swift
//
//  File.swift
//  
//
//  Created by Lowell Pence on 12/6/21.
//

import Foundation

struct Cell {
	enum State: Int {
		case neutral = 0
		case missed  = 1
		case hit     = 2
	}

	var state = State.neutral
	var isOccupied = false

	mutating func attack() -> Bool {
		state = isOccupied ? .hit : .missed
		return isOccupied
	}
}
